tree/ntuplebrowse/v7/inc/ROOT/RNTupleBrowser.hxx

```diff
@@ -0,0 +1,201 @@
+   
+   
+   RNTupleBrowser(TDirectory* directory=nullptr, int fUnitTest = 0);
+   ~RNTupleBrowser();
+   
+   /// Instantiates the RNTupleReader associated to the TDirectory.
+   void SetDirectory(TDirectory* directory);
+   /// Called when double-clicking the folder which appears afer double-clicking the .root file.
+   void Browse(TBrowser *b);
+   RNTupleReader* GetReaderPtr() const { return fReaderPtr.get(); }
+   int GetfUnitTest() const { return fUnitTest; }
+   void IncreasefUnitTest() { ++fUnitTest; }
+   Bool_t IsFolder() const { return kTRUE; }
+   
+   ClassDef(RNTupleBrowser,2)
+};
+
+// clang-format off
+/**
+\class ROOT::Experimental::RNTupleFieldElementFolder
+\ingroup NTupleBrowse
+\brief Is displayed in the TBrowser as a field with children.
+ 
+It represents an ntuple-field which has children. It is displayed in the TBrowser with a folder symbol. The subfields can be displayed by double-clicking this object, which calls void Browse(TBrowser *b)
+*/
+// clang-format on
+
+class RNTupleFieldElementFolder: public TNamed {
+private:
+   /// Pointer to the field it represents.
+   const Detail::RFieldBase*  fFieldPtr;
+   /// Pointer to the instance of RNTupleBrowser which created it through RBrowseVisitor::VisitField().
+   RNTupleBrowser*            fRNTupleBrowserPtr;
+   /// RNTupleReader used for reading fFieldPtr's subfields.
+   RNTupleReader*             fNtupleReaderPtr;
+public:
+   // ClassDef requires a constructor which can be called without any arguments. The constructor is never called with default arguments.
+   RNTupleFieldElementFolder(std::string name = "DummyName", const Detail::RFieldBase* FieldPtr = nullptr, RNTupleBrowser* ntplbPtr = nullptr): fFieldPtr{FieldPtr}, fRNTupleBrowserPtr{ntplbPtr}, fNtupleReaderPtr{ntplbPtr->GetReaderPtr()} {
+      fName = name;
+   }
+   
+   std::string GetFieldName() const { return static_cast<std::string>((const char*)fName); }
+   
+   /// Displays the field it represents in TBrowser. Called in RBrowseVisitor::VisitField()
+   void AddBrowse(TBrowser *b);
+   /// Displays subfields in TBrowser. Called when double-clicked.
+   void Browse(TBrowser *b);
+   
+   
+   Bool_t IsFolder() const { return true; }
+   ClassDef(RNTupleFieldElementFolder, 2)
+};
+
+// clang-format off
+/**
+\class ROOT::Experimental::RNTupleFieldElement
+\ingroup NTupleBrowse
+\brief Is displayed in the TBrowser as a field without children.
+    
+It represents an ntuple-field without children. If it represents a field with numerical data, double-clicking creates a TH1F-histogram.
+*/
+// clang-format on
+   
+class RNTupleFieldElement: public TNamed {
+private:
+   /// Data type of the field it represents.
+   fieldDatatype     fType;
+   /// Pointer to the instance of RNTupleBrowser which created it through RBrowseVisitor::VisitField().
+   RNTupleBrowser*   fRNTupleBrowserPtr;
+   /// RNTupleReader used for reading fFieldPtr's subfields.
+   RNTupleReader*    fReaderPtr;
```

Comment:
This one however can't be removed, because it calls the `GetView<...>`. From `fRNTupleBrowserPtr` one can only get the most recent RNTupleReader-pointer, which can be a different than the one whose field should be displayed. (As long as SetDirectory() isn't called, the most recent RNTupleReader in `fRNTupleBrowserPtr` doesn't change.)